mobile-browser-based-version/src/helpers/training/training_manager.js

```diff
@@ -145,105 +145,115 @@ export class TrainingManager {
     );
   }
 
+  async _modelFitData(model, trainingInformation, callbacks) {
+    const tensor = preprocessData(this.data, trainingInformation);
+
+    await model.fit(tensor, this.labels, {
+      initialEpoch: this.model.getUserDefinedMetadata().epoch,
+      epochs: trainingInformation.epochs ?? MANY_EPOCHS,
+      batchSize: trainingInformation.batchSize,
+      validationSplit: trainingInformation.validationSplit,
+      shuffle: true,
+      callbacks: callbacks,
+    });
+  }
+
+  async _modelFitDataBatchWise(model, trainingInformation, callbacks) {
+    // Creation of Dataset objects for training
+    const trainData = tf.data
+      .generator(
+        datasetGenerator(
+          this.data,
+          this.labels,
+          0,
+          this.data.shape[0] * (1 - trainingInformation.validationSplit),
+          trainingInformation
+        )
+      )
+      .batch(trainingInformation.batchSize);
+
+    const valData = tf.data
+      .generator(
+        datasetGenerator(
+          this.data,
+          this.labels,
+          Math.floor(
+            this.data.shape[0] * (1 - trainingInformation.validationSplit)
+          ),
+          this.data.shape[0],
+          trainingInformation
+        )
+      )
+      .batch(trainingInformation.batchSize);
+
+    await model.fitDataset(trainData, {
+      epochs: trainingInformation.epochs,
+      validationData: valData,
+      callbacks: callbacks,
+    });
+  }
+
   async _trainLocally() {
     const info = this.task.trainingInformation;
 
-    if (info.batchwise) {
+    if (info.batchwisePreprocessing) {
       console.log(
         'Memory efficient training mode is used, data preprocessing is executed batch wise'
       );
-      // Creation of Dataset objects for training
-      const trainData = tf.data
-        .generator(
-          datasetGenerator(
-            this.data,
-            this.labels,
-            0,
-            this.data.shape[0] * (1 - info.validationSplit),
-            info
-          )
-        )
-        .batch(info.batchSize);
-
-      const valData = tf.data
-        .generator(
-          datasetGenerator(
-            this.data,
-            this.labels,
-            Math.floor(this.data.shape[0] * (1 - info.validationSplit)),
-            this.data.shape[0],
-            info
-          )
-        )
-        .batch(info.batchSize);
-      await this.model.fitDataset(trainData, {
-        epochs: info.epochs,
-        validationData: valData,
-        callbacks: {
-          onEpochEnd: async (epoch, logs) => {
-            this.trainingInformant.updateGraph(
-              epoch + 1,
-              (logs.val_acc * 100).toFixed(2),
-              (logs.acc * 100).toFixed(2)
-            );
-            console.log(
-              `EPOCH (${epoch + 1}):
+      await this._modelFitDataBatchWise(this.model, info, {
+        onEpochEnd: async (epoch, logs) => {
```

Comment:
The onEpochEnd is the same for ._modelFitDataBatchWis and .modelFit, you could turn the onEpochEnd and make a function and pass it to both to reduce code duplication.